monai/apps/deepgrow/transforms.py

```diff
@@ -0,0 +1,623 @@
+                signal[i] = gaussian_filter(signal[i], sigma=self.sigma)
+                signal[i] = (signal[i] - np.min(signal[i])) / (np.max(signal[i]) - np.min(signal[i]))
+        return signal
+
+    def _apply(self, image, guidance):
+        if not self.batched:
+            signal = self._get_signal(image, guidance)
+            return np.concatenate([image, signal], axis=0)
+
+        images = []
+        for i, g in zip(image, guidance):
+            i = i[0 : 0 + self.number_intensity_ch, ...]
+            signal = self._get_signal(i, g)
+            images.append(np.concatenate([i, signal], axis=0))
+        return images
+
+    def __call__(self, data):
+        image = data[self.image]
+        guidance = data[self.guidance]
+
+        data[self.image] = self._apply(image, guidance)
+        return data
+
+
+class FindDiscrepancyRegionsd(Transform):
+    def __init__(self, label="label", pred="pred", discrepancy="discrepancy", batched=True):
+        self.label = label
+        self.pred = pred
+        self.discrepancy = discrepancy
+        self.batched = batched
+
+    @staticmethod
+    def disparity(label, pred):
+        label = (label > 0.5).astype(np.float32)
+        pred = (pred > 0.5).astype(np.float32)
+        disparity = label - pred
+
+        pos_disparity = (disparity > 0).astype(np.float32)
+        neg_disparity = (disparity < 0).astype(np.float32)
+        return [pos_disparity, neg_disparity]
+
+    def _apply(self, label, pred):
+        if not self.batched:
+            return self.disparity(label, pred)
+
+        disparity = []
+        for la, pr in zip(label, pred):
+            disparity.append(self.disparity(la, pr))
+        return disparity
+
+    def __call__(self, data):
+        label = data[self.label]
+        pred = data[self.pred]
+
+        data[self.discrepancy] = self._apply(label, pred)
+        return data
+
+
+class AddRandomGuidanced(Randomizable, Transform):
+    def __init__(self, guidance="guidance", discrepancy="discrepancy", probability="probability", batched=True):
+        self.guidance = guidance
+        self.discrepancy = discrepancy
+        self.probability = probability
+        self.batched = batched
+
+    def randomize(self, data=None):
+        pass
+
+    @staticmethod
+    def find_guidance(discrepancy):
+        distance = distance_transform_cdt(discrepancy).flatten()
+        probability = np.exp(distance) - 1.0
+        idx = np.where(discrepancy.flatten() > 0)[0]
+
+        if np.sum(discrepancy > 0) > 0:
+            seed = np.random.choice(idx, size=1, p=probability[idx] / np.sum(probability[idx]))
+            dst = distance[seed]
+
+            g = np.asarray(np.unravel_index(seed, discrepancy.shape)).transpose().tolist()[0]
+            g[0] = dst[0]
+            return g
+        return None
+
+    @staticmethod
+    def add_guidance(discrepancy, probability):
+        will_interact = np.random.choice([True, False], p=[probability, 1.0 - probability])
```

Comment:
random operations should be done inside ``randomize`` function and use ``self.R``

Reply:
it didn't work in case of interactions.. may be i can add a note to get it under randomize function in future